1971.py

```diff
@@ -0,0 +1,20 @@
+class Solution:
+    def validPath(self, n: int, edges: List[List[int]], start: int, end: int) -> bool:
+        seen, graph = set(), defaultdict(list)
+
+        for s, e in edges:
+            graph[s].append(e)
+            graph[e].append(s)
+
+        queue = deque([start])
+
+        while queue:
+            curr = queue.pop()
+
+            if curr == end:
+                return True
+            elif curr in graph and curr not in seen:
```

Comment:
Your solution may have been accepted by LeetCode, but I can tell that you are confused by the meaning and purpose of this `seen` variable.

In BFS, we use this `seen` variable to denote that a vertex has been added to the queue, such that the same vertex **does not get added more than once and avoids infinite loop**. Here in line 17, you are adding the neighbors of `curr` to the queue, not `curr` itself. You shouldn't be adding `curr` to `seen` here; instead, you should add every vertex in `graph[curr]` to `seen` because you just added them to the queue.

By the same reasoning, in line 10, you should add `start` to `seen` because that's where vertex `start` is seen for the first time and that's where it is added to the queue.

Note that we name this variable to `seen` not because it's cool or something. It literally means a vertex is seen, discovered, and just got added to the queue. It does not mean something like "the vertex has been visited or processed", which seems to be the case in your mind right now.

Regarding the different between `visited` and `seen`, I mentioned it a few months ago [here](https://github.com/bunnygod/leetcode-python/pull/237#discussion_r680713327).